integration_tests/e2e/booking.cy.ts

```diff
@@ -0,0 +1,48 @@
+import premisesFactory from '../../server/testutils/factories/premises'
+import bookingFactory from '../../server/testutils/factories/booking'
+import BookingPage from '../pages/booking'
+import PremisesPage from '../pages/premises'
+import Page from '../pages/page'
+
+context('Booking', () => {
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubSignIn')
+    cy.task('stubAuthUser')
+  })
+
+  it('should show booking form', () => {
+    const booking = bookingFactory.build()
+    const premises = premisesFactory.buildList(5)
+    cy.task('stubBooking', booking)
+    cy.task('stubPremises', premises)
+
+    // Given I am signed in
+    cy.signIn()
+
+    // When I visit the booking page
+    const page = BookingPage.visit()
+
+    // Then I should see the booking form and enter details
+    page.getLabel('CRN')
+    page.getByInputByIdAndEnterDetails('CRN', 'A123456')
+
+    page.getLegend('What is the arrival date?')
+    page.arrivalDay().type('21')
+    page.arrivalMonth().type('07')
+    page.arrivalYear().type('22')
+
+    page.getLegend('What is the expected departure date?')
+    page.expectedDepartureDay().type('30')
+    page.expectedDepartureMonth().type('12')
+    page.expectedDepartureYear().type('22')
+
+    page.getLabel('Key Worker')
+    page.getByInputByIdAndEnterDetails('keyWorker', 'John Doe')
```

Comment:
I'd prefer it if we could hide the implementation in the page object and pass in a booking that has been created by a factory, so somthing like:

```typescript
const booking = bookingFactory.build()
page.completeBookingForm(booking)
```

Then the page object would have a `completeBookingForm` method that looked like this:

```typescript
this.getLabel('CRN')
this.getByInputByIdAndEnterDetails('CRN', booking.crn)

this.getLegend('What is the arrival date?')
this.arrivalDay().type(booking.arrivalDate.day())
this.arrivalMonth().type(booking.arrivalDate.month())
this.arrivalYear().type(booking.arrivalDate.year())

this.getLegend('What is the expected departure date?')
this.expectedDepartureDay().type(booking.departureDate.day())
this.expectedDepartureMonth().type(booking.departureDate.month())
this.expectedDepartureYear().type(booking.departureDate.year())

this.getLabel('Key Worker')
this.getByInputByIdAndEnterDetails('keyWorker', booking.keyWorker)
```

Reply:
Yeah I agree that your suggested approach is more readable